### PR TITLE
Feature/pelagos 3496 add login form

### PR DESCRIPTION
**Pull request by mickel1138**
~~This code doesn't work yet, but it's enough to get you "authenticated" and get you into all the other applications.~~

The authenticator is now working. See comments on: https://symfonycasts.com/screencast/symfony3-security/user-provider

### CODE REVIEW

**Comment by fightingtexasaggie** [comment]
Why you merging into develop?!

File: src/Pelagos/Bundle/AppBundle/Security/LoginFormAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return $theUser;
+    }
+
+    public function checkCredentials($credentials, UserInterface $user)
+    {
+        $password = $credentials['_password'];
+        if ($password == 'CHECK PASSWORD HERE!') {
+            return true;
+        }
+        return true; //Normally false, be this is TEST!
```
Comment (fightingtexasaggie) [inline src/Pelagos/Bundle/AppBundle/Security/LoginFormAuthenticator.php]: https://www.youtube.com/watch?v=a6iW-8xPw3k

File: src/Pelagos/Bundle/AppBundle/Security/PelagosEncoder.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        return sha1($raw . $salt, true);
+    }
+
+    public function isPasswordValid($encoded, $raw, $salt)
+    {
+        return $encoded === $this->encodePassword($raw, $salt);
+    }
+
+}
```
Comment (fightingtexasaggie) [inline src/Pelagos/Bundle/AppBundle/Security/PelagosEncoder.php]: add newline

File: src/Pelagos/Bundle/AppBundle/Form/LoginForm.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                'label' => 'Password',
+                'required' => true,
+                'attr' => array(
+                    'placeholder' => 'Password',
+                    'class' => 'form-control',
+                ),
+            
+            ));
+    }
+}
```
Comment (snguyen1) [inline src/Pelagos/Bundle/AppBundle/Form/LoginForm.php]: new line

File: app/config/config.yml [hunk 4]
```diff
@@ -20,6 +20,7 @@ parameters:
 framework:
     #esi:             ~
     #translator:      { fallbacks: ["%locale%"] }
+    csrf_protection: ~
```
Comment (praneethpr) [inline app/config/config.yml]: I thought symfony forms had default CSRF protection. Good that we are implementing now

File: src/Pelagos/Entity/Account.php [hunk 5]
```diff
@@ -547,4 +548,9 @@ public function getSalt()
         }
         return $this->password->getSalt();
     }
+    
+    public function isEqualTo(UserInterface $user)
```
Comment (praneethpr) [inline src/Pelagos/Entity/Account.php]: No doc block

File: src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
```
Comment (praneethpr) [inline src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php]: Can remove extra lines between use statements

Comment (mickel1138) [reply src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php]: Normally we separate use statements by namespace.

File: app/config/config.yml (hunk 4 above)
Comment (mickel1138) [reply app/config/config.yml]: I'm not sure how well it's implemented, but I had to add a hidden field to this form in html myself.

**Comment by praneethpr** [comment]
> Layout broken (only happening to me). We decide to fix CSS on later pull requests

The form doesn't align properly for me as well, so i don't think its only you

**Pull request merged**